### ISSUE TITLE
Feature suggestion: Support GFM YAML metadata block

### ISSUE THREAD

**Issue by zmwangx** [jrblevin/markdown-mode#66]
Some Markdown implementations, e.g., GFM and Pandoc, support YAML metadata blocks. For GFM (according to testing, since it seems undocumented) the YAML block must start at the beginning at the document (even blank lines before it are not allowed) and be wrapped in `---` (*exactly* three dashes) at the opening and closing, so an example document would be

```markdown
---
title: README
author: Zhiming Wang
date: January 8, 2016
---

This is an example GFM document with a YAML metadata block.
```

and this will be rendered like

<img width="456" alt="screen shot 2016-01-08 at 12 21 10 pm" src="https://cloud.githubusercontent.com/assets/4149852/12208559/5c1b1326-b602-11e5-88ed-9dadfca1160f.png">

Note that the blank line after the closing `---` line is optional.

(Pandoc, on the other hand, has very lax rules for YAML metadata blocks, which allows them to be placed anywhere in the document, and allows closing with `...` instead of `---`. But Pandoc Markdown doesn't need to be supported.)

Currently, the line immediately above the closing `---` is interpreted as a setext header, which looks rather weird (especially if you have a real header):

<img width="370" alt="screen shot 2016-01-08 at 12 22 42 pm" src="https://cloud.githubusercontent.com/assets/4149852/12208658/050ee26e-b603-11e5-8c0a-49b21c6027f9.png">

Would you please consider supporting the metadata block? I suppose the block itself could be highlighted just like a fenced code block, and I suppose it shouldn't be too hard to adapt the parsing code of fenced code blocks to this particular block. Thanks.

**Comment by jrblevin** [comment]
Supporting this would be fine with me. I'll add this to the todo list. If anyone would like to submit a patch for this, that would be helpful. The existing Pandoc and MultiMarkdown metadata parsing functions should serve as good examples.

**Comment by zmwangx** [comment]
Cool, thanks. I may give it a try myself when I have time.

**Comment by cosmicexplorer** [comment]
I made a pull request that seems to fix this. It's unfortunate that YAML is actually kinda hard to parse in its entirety; this doesn't attempt to highlight outside of simple `key: value` pairs, even though [more complex constructions](http://pandoc.org/README.html#extension-yaml_metadata_block) are used too. This should work github pages editing, though.

**Comment by syohex** [comment]
I have merged @cosmicexplorer PR(#91). Please check latest version. If you have still problems, please reopen the issue and report us.

**Comment by khatchad** [comment]
This isn't working in pandoc 2.5.
